src/plugins/data_importer/public/components/data_importer_app.tsx

```diff
@@ -147,39 +150,72 @@ export const DataImporterPluginApp = ({
   };
 
   const previewData = async () => {
-    if (importType === IMPORT_CHOICE_FILE) {
-      if (inputFile) {
-        const fileExtension = extname(inputFile.name);
-        setIsLoadingPreview(true);
-        try {
-          const response = await previewFile({
-            http,
-            file: inputFile,
-            createMode,
-            fileExtension,
-            indexName: indexName!,
-            previewCount: config.filePreviewDocumentsCount,
-            delimiter,
-            selectedDataSourceId: dataSourceId,
-          });
-          setIsLoadingPreview(false);
-          if (response) {
-            setFilePreviewData(response);
-            notifications.toasts.addSuccess(
-              i18n.translate('dataImporter.previewSuccess', {
-                defaultMessage: `Preview successful`,
-              })
-            );
-          } else {
-            notifications.toasts.addDanger(
-              i18n.translate('dataImporter.previewFailed', {
-                defaultMessage: `Preview failed`,
-              })
-            );
-          }
-        } catch (error) {
-          setIsLoadingPreview(false);
+    if (importType === IMPORT_CHOICE_FILE && inputFile) {
+      const fileExtension = extname(inputFile.name);
+      setIsLoadingPreview(true);
+      try {
+        const response = await previewFile({
+          http,
+          file: inputFile,
+          createMode,
+          fileExtension,
+          indexName: indexName!,
+          previewCount: config.filePreviewDocumentsCount,
+          delimiter,
+          selectedDataSourceId: dataSourceId,
+        });
+        setIsLoadingPreview(false);
+        if (response) {
+          setFilePreviewData(response);
+          notifications.toasts.addSuccess(
+            i18n.translate('dataImporter.previewSuccess', {
+              defaultMessage: `Preview successful`,
+            })
+          );
+          setSelectedTab('preview');
+        } else {
+          notifications.toasts.addDanger(
+            i18n.translate('dataImporter.previewFailed', {
+              defaultMessage: `Preview failed`,
+            })
+          );
+        }
+      } catch (error) {
+        setIsLoadingPreview(false);
+      }
+    } else if (importType === IMPORT_CHOICE_TEXT && inputText) {
+      const fileExtension = `.${dataType}`;
+      const textBlob = new Blob([inputText], { type: 'text/plain' });
```

Comment:
Could we extract this text -> File object conversion to a separate utility function and add tests here?